extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -291,6 +349,316 @@ export class MigrationStateModel implements Model, vscode.Disposable {
+		try {
+			const serverInfo = await azdata.connection.getServerInfo(this.sourceConnectionId);
+			const machineName = (<any>serverInfo)['machineName'];		// get actual machine name instead of whatever the user entered as the server name (e.g. DESKTOP-xxx instead of localhost)
+
+			const response = (await this.migrationService.getSkuRecommendations(
+				this._skuRecommendationPerformanceLocation,
+				this._performanceDataQueryIntervalInSeconds,
+				this._recommendationTargetPlatforms.map(p => p.toString()),
+				machineName,
+				this._skuTargetPercentile,
+				this._skuScalingFactor,
+				this._defaultDataPointStartTime,
+				this._defaultDataPointEndTime,
+				this._skuEnablePreview,
+				this._databaseAssessment))!;
+			this._skuRecommendationApiResponse = response;
+
+			// clone list of databases currently being assessed and store them, so that if the user ever changes the list we can refresh new recommendations
+			this._skuRecommendationRecommendedDatabaseList = this._databaseAssessment.slice();
+
+			console.log('sqlinstancerequirements: ');
+			console.log(this._skuRecommendationApiResponse.instanceRequirements);
+
+			if (response?.sqlDbRecommendationResults || response?.sqlMiRecommendationResults || response?.sqlVmRecommendationResults) {
+				this._skuRecommendationResults = {
+					recommendations: {
+						sqlDbRecommendationResults: response?.sqlDbRecommendationResults ?? [],
+						sqlMiRecommendationResults: response?.sqlMiRecommendationResults ?? [],
+						sqlVmRecommendationResults: response?.sqlVmRecommendationResults ?? [],
+						instanceRequirements: response?.instanceRequirements
+					},
+					// recommendationError:
+				};
+			} else {
+				this._skuRecommendationResults = {
+					recommendations: {
+						sqlDbRecommendationResults: [],
+						sqlMiRecommendationResults: [],
+						sqlVmRecommendationResults: [],
+						instanceRequirements: response?.instanceRequirements
+					},
+					// recommendationError:
+				};
+			}
+
+		} catch (error) {
+			console.log(error);
+
+			this._skuRecommendationResults = {
+				recommendations: {
+					sqlDbRecommendationResults: this._skuRecommendationApiResponse?.sqlDbRecommendationResults ?? [],
+					sqlMiRecommendationResults: this._skuRecommendationApiResponse?.sqlMiRecommendationResults ?? [],
+					sqlVmRecommendationResults: this._skuRecommendationApiResponse?.sqlVmRecommendationResults ?? [],
+					instanceRequirements: this._skuRecommendationApiResponse?.instanceRequirements
+				},
+				recommendationError: error
+			};
+		}		// Generating all the telemetry asynchronously as we don't need to block the user for it.
+		this.generateSkuRecommendationTelemetry().catch(e => console.error(e));
+
+		return this._skuRecommendationResults;
+	}
+
+	private async generateSkuRecommendationTelemetry(): Promise<void> {
+		try {
+
+			this._skuRecommendationResults?.recommendations?.sqlMiRecommendationResults?.forEach(resultItem => {
+				// Send telemetry for recommended MI SKU
```

Comment:
how many telemetry events do you expect to see in each of these events, will it be 1 or 2 or many more?

Reply:
For now it’s just one sku recommendation/ one event.